### PR TITLE
Feature 420 implement get postulations by project

### PR DESCRIPTION
**Pull request by rodrigo-baldiviezo-jalafund**
[AB#420](https://dev.azure.com/funjaladev/DevLevel3/_sprints/backlog/DevLevel3%20Team/DevLevel3/Sprint%207?workitem=420)

# Overview

Implement Get operation

### CODE REVIEW

**Review by niver-copa-jalafund — DISMISSED** [review]

**Review by lina-ramirez-jalafund — DISMISSED** [review]

File: src/api/Jalasoft.TeamUp.Resumes.API/Controllers/GetPostulationsByProjectId.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(Postulation[]), Description = "Successful response")]
+        [OpenApiResponseWithoutBody(statusCode: HttpStatusCode.NotFound, Description = "Resource not found")]
+        public IActionResult Run(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "v1/postulations")] HttpRequest req)
+        {
+            try
+            {
+                req.Query.TryGetValue("projectId", out StringValues projectId);
+                var postulations = this.postulationsService.GetPostulationsByProjectId(projectId);
+                return new OkObjectResult(postulations);
```
Comment (Raul-Gamarra-jalafund) [inline src/api/Jalasoft.TeamUp.Resumes.API/Controllers/GetPostulationsByProjectId.cs:42]: please add not found response

File: tests/api/Jalasoft.TeamUp.Resumes.API.Tests/GetPostulationsByProjectIdTests.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            // Arrange
+            var request = this.mockHttpContext.Request;
+            this.mockService.Setup(service => service.GetPostulationsByProjectId("1")).Returns(new Postulation[2]);
+
+            // Act
+            var response = this.getPostulationsByProjectId.Run(request);
+
+            // Assert
+            var okObjectResult = Assert.IsType<OkObjectResult>(response);
+            Assert.IsType<Postulation[]>(okObjectResult.Value);
```
Comment (Raul-Gamarra-jalafund) [inline tests/api/Jalasoft.TeamUp.Resumes.API.Tests/GetPostulationsByProjectIdTests.cs:72]: please check that the response code is 200 here

File: tests/api/Jalasoft.TeamUp.Resumes.API.Tests/GetPostulationsByProjectIdTests.cs [hunk 3]
```diff
@@ -52,5 +86,20 @@ public void GetPostulationsByProjectId_InvalidResponse_Error500()
             var objectResult = Assert.IsType<ObjectResult>(response);
             Assert.Equal(500, objectResult.StatusCode);
         }
+
+        [Fact]
+        public void GetPostulationsByProjectId_InvalidProjectId_BadRequest()
+        {
+            // Arrange
+            var request = this.mockHttpContext.Request;
+            this.mockService.Setup(service => service.GetPostulationsByProjectId(It.IsAny<string>())).Throws(new ResumesException(ResumesErrors.NotFound));
```
Comment (Raul-Gamarra-jalafund) [inline tests/api/Jalasoft.TeamUp.Resumes.API.Tests/GetPostulationsByProjectIdTests.cs]: the service only returns an empty array the api throws the error

Comment (rodrigo-baldiviezo-jalafund) [reply tests/api/Jalasoft.TeamUp.Resumes.API.Tests/GetPostulationsByProjectIdTests.cs]: done

**Review by Raul-Gamarra-jalafund — APPROVED** [review]

**Review by niver-copa-jalafund — APPROVED** [review]

**Review by lina-ramirez-jalafund — APPROVED** [review]

**Pull request merged**